src/transformers/models/idefics/processing_idefics.py

```diff
@@ -149,7 +149,7 @@ def __init__(self, image_processor, tokenizer=None, image_size=224, add_end_of_u
     def __call__(
         self,
         prompts: Union[List[TextInput], List[List[TextInput]]],
-        padding: Union[bool, str, PaddingStrategy] = False,
+        padding: Union[bool, str, PaddingStrategy] = "longest",
```

Comment:
I don't think we should change the default here for two reasons:
* It doesn't match the default behaviour for most processing classes
* It changes the default behaviour, which can be considered a breaking change

Reply:
> bri25yu

You may have tagged the wrong person 🙃

>It doesn't match the default behaviour for most processing classes

Does the idefics model support non-padded inputs? From my understanding of the [original issue](https://huggingface.co/HuggingFaceM4/idefics-9b-instruct/discussions/11), it seems they desire expect some padding even when the argument is not passed.

>It changes the default behaviour, which can be considered a breaking change

I'm not sure if this is a bug, but the default behavior appears to have been inaccurate to begin with. Even if the user passes in `padding=False`, lines 347-354 seems to be forcibly padding the text input to the maximum sequence length anyways.

Reply:
> You may have tagged the wrong person 🙃

Woops, yes, sorry about that

> Does the idefics model support non-padded inputs? From my understanding of the [original issue](https://huggingface.co/HuggingFaceM4/idefics-9b-instruct/discussions/11), it seems they desire expect some padding even when the argument is not passed.

No, but no models support non-padded inputs if batch_size > 1 and the input sequences are of different lengths, but all processors and tokenizers do not pad the inputs by default. 

> I'm not sure if this is a bug, but the default behavior appears to have been inaccurate to begin with. Even if the user passes in padding=False, lines 347-354 seems to be forcibly padding the text input to the maximum sequence length anyways.

In this case, the forcible padding when `padding=False` should be removed

Reply:
> In this case, the forcible padding when padding=False should be removed

Yes, this behavior was removed as part of this PR.

I have modified the PR to default to `padding=False`, and the unit tests to explicitly specify `padding='longest'`. I guess my only concern was that by changing this behavior, a user which was relying on the default behavior would find their code broken overnight.

Reply:
> I guess my only concern was that by changing this behavior, a user which was relying on the default behavior would find their code broken overnight.

@byi8220 Hmm, yes, this is tricky and that's a good point. OK, in this case, I think your original solution of setting `padding='longest'` as default is best, ideally with a comment linking to this issue to explain why the default is different and adding a description for the `False` option in the docstring

Reply:
Done